syntax = "proto3";

package profile;

option java_package = "com.example.findHataProposalServer";
option java_generic_services = true;

service VectorizationService {
    rpc VectorizeProposal(Proposal) returns (stream TextVector) {}
    rpc VectorizeRequest(Request) returns (stream TextVector) {}
}

message Proposal {
    string title = 1;
    string description = 2;
    string location = 3;
}

message Request {
    string request = 1;
}


message TextVector {
    repeated double vector = 1;
    bool is_negative = 2;
}
